spirv_glsl.cpp

```diff
@@ -3141,6 +3149,9 @@ string CompilerGLSL::access_chain(uint32_t base, const uint32_t *indices, uint32
 			throw CompilerError("Cannot subdivide a scalar value!");
 	}
 
+	if (mtx_needs_transp)
+		expr = transpose(expr);
+
```

Comment:
Another case is that transposing in the access chain doesn't support OpStore to a transposed matrix.
Although, I'm not sure how important this case is. I'd like to have an exception being thrown that unless the access chain is accessing storage class Input (read-only), UniformConstant (UBO) or PushConstant, row-major is flagged as currently not supported.